TestErrorSpam/setup fails on every PR after k8s-dependencies bump (proxy.go RoundTripper cast)

### What Happened?

`TestErrorSpam/setup` is failing on every PR's prow integration tests since the k8s-dependencies bump in [#22876](https://github.com/kubernetes/minikube/pull/22876) (commit `f711fb429`) merged on 2026-04-24. The failure is reproducible on at minimum docker, containerd, kvm and crio runtimes across both x86 and arm.

The test asserts that `minikube start` produces no unexpected stderr. After the dep bump, every `minikube start` now emits:

```
E0425 20:01:48.250329   63697 proxy.go:152] Error while casting RoundTripper (of type *transport.trackedTransport) to *http.Transport : false
E0425 20:01:48.368989   63697 proxy.go:152] Error while casting RoundTripper (of type *transport.trackedTransport) to *http.Transport : false
E0425 20:01:48.369055   63697 proxy.go:152] Error while casting RoundTripper (of type *transport.trackedTransport) to *http.Transport : false
```

`TestErrorSpam` flags the lines as unexpected stderr and fails:

```
--- FAIL: TestErrorSpam/setup (14.80s)
    error_spam_test.go:100: unexpected stderr: "E0425 20:01:48.250329   63697 proxy.go:152] Error while casting RoundTripper (of type *transport.trackedTransport) to *http.Transport : false"
    ...
```

### Attach the log file

Any PR opened against current `master` reproduces. Verified across at least:

- #22862 (this is what surfaced the regression): https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/minikube/22862/pull-minikube-docker-docker-linux-x86/2048128180010618880
- #22895 (an unrelated kubevirt addon bump, only TestErrorSpam/setup fails): https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/minikube/22895/pull-minikube-docker-containerd-linux-x86/2047970412591058944

The failure is the sole `Fail` in test_summary.json on most lanes, with `Pass` numbers in the high 300s, so the rest of the integration suite is healthy. Master itself almost certainly fails the same way; this is blocking PR merges across the repo.

### Operating System

None

### Driver

None

--- a/pkg/minikube/proxy/proxy.go
+++ b/pkg/minikube/proxy/proxy.go
@@ -27,6 +27,7 @@ import (
 
 	"errors"
 
+	utilnet "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/client-go/rest"
 	"k8s.io/klog/v2"
 	"k8s.io/minikube/pkg/minikube/config"
@@ -145,15 +146,32 @@ func UpdateTransport(cfg *rest.Config) *rest.Config {
 		if wt != nil {
 			rt = wt(rt)
 		}
-		if ht, ok := rt.(*http.Transport); ok {
-			ht.Proxy = nil
-			rt = ht
-		} else {
-			klog.Errorf("Error while casting RoundTripper (of type %T) to *http.Transport : %v", rt, ok)
+		if t := wrappedHTTPTransport(rt); t != nil {
+			t.Proxy = nil
+		} else if rt != nil {
+			klog.Errorf("transport %T could not be unwrapped to *http.Transport; proxy stripping was skipped", rt)
 		}
 		return rt
 	}
 	return cfg
+}
+
+// wrappedHTTPTransport returns the *http.Transport at the bottom of any
+// RoundTripperWrapper chain wrapping rt, or nil if the chain does not end
+// in one (or rt itself is nil). See
+// https://github.com/kubernetes/minikube/issues/22896 for context.
+func wrappedHTTPTransport(rt http.RoundTripper) *http.Transport {
+	cur := rt
+	for {
+		switch t := cur.(type) {
+		case *http.Transport:
+			return t
+		case utilnet.RoundTripperWrapper:
+			cur = t.WrappedRoundTripper()
+		default:
+			return nil
+		}
+	}
 }
 
 // SetDockerEnv sets the proxy environment variables in the docker environment.

--- a/pkg/minikube/proxy/proxy_test.go
+++ b/pkg/minikube/proxy/proxy_test.go
@@ -22,6 +22,7 @@ import (
 	"os"
 	"testing"
 
+	utilnet "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/client-go/rest"
 )
 
@@ -188,6 +189,24 @@ func TestExcludeIP(t *testing.T) {
 	}
 }
 
+// fakeRoundTripperWrapper satisfies utilnet.RoundTripperWrapper for tests
+// that need to simulate the way client-go 0.36's *trackedTransport (or any
+// future wrapping layer) interposes between WrapTransport callers and the
+// underlying *http.Transport.
+type fakeRoundTripperWrapper struct{ rt http.RoundTripper }
+
+func (f *fakeRoundTripperWrapper) RoundTrip(req *http.Request) (*http.Response, error) {
+	return f.rt.RoundTrip(req)
+}
+
+func (f *fakeRoundTripperWrapper) WrappedRoundTripper() http.RoundTripper { return f.rt }
+
+// Compile-time assertion: fakeRoundTripperWrapper must satisfy
+// utilnet.RoundTripperWrapper so the test exercises the same interface the
+// helper walks at runtime. Future client-go interface changes will surface
+// here as a build failure rather than a silently irrelevant test.
+var _ utilnet.RoundTripperWrapper = (*fakeRoundTripperWrapper)(nil)
+
 func TestUpdateTransport(t *testing.T) {
 	t.Run("new", func(t *testing.T) {
 		rcBefore := rest.Config{}
@@ -221,6 +240,36 @@ func TestUpdateTransport(t *testing.T) {
 		rt := c.WrapTransport(nil)
 		if rt != nil {
 			t.Fatalf("Expected RoundTripper nil for invocation WrapTransport(nil)")
+		}
+	})
+	t.Run("wrapped", func(t *testing.T) {
+		inner := &http.Transport{Proxy: http.ProxyFromEnvironment}
+		wrapped := &fakeRoundTripperWrapper{rt: inner}
+
+		rc := rest.Config{}
+		c := UpdateTransport(&rc)
+		result := c.WrapTransport(wrapped)
+
+		if result != wrapped {
+			t.Fatalf("expected wrapper to be returned unchanged, got %T", result)
+		}
+		if inner.Proxy != nil {
+			t.Errorf("expected inner transport's Proxy to be nil, got non-nil")
+		}
+	})
+	t.Run("proxy-stripping", func(t *testing.T) {
+		inner := &http.Transport{Proxy: http.ProxyFromEnvironment}
+
+		rc := rest.Config{}
+		c := UpdateTransport(&rc)
+		result := c.WrapTransport(inner)
+
+		resultHT, ok := result.(*http.Transport)
+		if !ok {
+			t.Fatalf("expected *http.Transport from no-wrap path, got %T", result)
+		}
+		if resultHT.Proxy != nil {
+			t.Errorf("expected Proxy to be nil after UpdateTransport, got non-nil")
 		}
 	})
 }

